### PR TITLE
Test  instanceof Passport instead of more restrictive SelfValidatingPassport

### PR DESCRIPTION
**Pull request by TristanPouliquen**
…assport

### CODE REVIEW

File: Security/Authenticator/JWTAuthenticator.php [hunk 1]
```diff
@@ -240,7 +240,7 @@ protected function loadUser(array $payload, string $identity): UserInterface
 
     public function createAuthenticatedToken(PassportInterface $passport, string $firewallName): TokenInterface
     {
-        if (!$passport instanceof SelfValidatingPassport) {
+        if (!$passport instanceof Passport) {
             throw new \LogicException(sprintf('Expected "%s" but got "%s".', SelfValidatingPassport::class, get_debug_type($passport)));
```
Comment (fd6130) [inline Security/Authenticator/JWTAuthenticator.php]: You forgot to change this to `Passport::class`.

_Edit:_
Already updated before i write this.

**Review by fd6130 — COMMENTED** [review]
Seems nice to me.

**Review by chalasr — APPROVED** [review]
Excellent 👍

File: Security/Authenticator/JWTAuthenticator.php [hunk 2]
```diff
@@ -240,8 +240,8 @@ protected function loadUser(array $payload, string $identity): UserInterface
 
     public function createAuthenticatedToken(PassportInterface $passport, string $firewallName): TokenInterface
```
Comment (fd6130) [inline Security/Authenticator/JWTAuthenticator.php:241]: >  (*PassportInterface are deprecated in Symfony 5.4)

What about this argument type here?

Comment (TristanPouliquen) [reply Security/Authenticator/JWTAuthenticator.php:241]: I think @chalasr was talking about `UserPassportInterface` here, and not the generic `PassportInterface` (the * before makes me say this)

Comment (fd6130) [reply Security/Authenticator/JWTAuthenticator.php:241]: Oh.. i didn't thought about it. But I think it is PassportInterface? You can check from here https://github.com/symfony/symfony/blob/5.4/CHANGELOG-5.4.md

Let's wait him for his opinion about this method argument type change.

Comment (TristanPouliquen) [reply Security/Authenticator/JWTAuthenticator.php:241]: Of what I've seen the whole `createAuthenticatedToken` is going to be deprecated in favor of the simple `createToken` method in any case :)

Comment (chalasr) [reply Security/Authenticator/JWTAuthenticator.php:241]: This method must remain the same for compatibility. Both `PassportInterface` and `createAuthenticatedToken()` are deprecated and removed in Symfony 6.0. 
Note that the reasoning for deprecating `PassportInterface` is that passports are not an extension point, one should use custom badges instead.

**Comment by chalasr** [comment]
Thanks Tristan.

**Pull request merged**